#4197 - Minor improvements to time conductor history

mariuszr opened the pull request:
## Overview
Addresses #4197 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)? However, the changes have been adapted to the current programming convention used in the specific files.
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this change backwards compatible? Will users need to change how they are calling the API, or how they've extended core plugins such as Tables or Plots?

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

mariuszr commented:
Hi @davetsay / @jvigliotta, please review.

jvigliotta commented on platform/commonUI/formats/src/UTCTimeFormat.js:
```diff
@@ -55,13 +55,21 @@ define([
      * in the array.
      */
-    UTCTimeFormat.prototype.format = function (value) {
+    UTCTimeFormat.prototype.format = function (value, formatId) {
```
can we change this to formatString instead of formatId? I'd like to pass the format string in here, check if it exists in the DATE_FORMATS array and use that if it does, and default to DATE_FORMAT if not or if no string is passed in. This way the developer can pass a format string in and it's more clear what format they're choosing.

jvigliotta replied:
can we change this to formatString instead of formatId? I'd like to pass the format string in here, check if it exists in the DATE_FORMATS array and use that if it does, and default to DATE_FORMAT if not or if no string is passed in. This way the developer can pass a format string in and it's more clear what format they're choosing.

jvigliotta commented on src/plugins/timeConductor/ConductorHistory.vue:
```diff
@@ -263,7 +263,7 @@ export default {
             }).formatter;
 
-            return (isNegativeOffset ? '-' : '') + formatter.format(time);
+            return (isNegativeOffset ? '-' : '') + formatter.format(time, 2);
```
instead of 2, let's pass in the string: `"YYYY-MM-DD HH:mm:ss"`

jvigliotta commented on src/utils/duration.js:
```diff
@@ -39,6 +40,16 @@ function toDoubleDigits(num) {
+    const m = Math.floor(normalizeAge((ms % ONE_HOUR) / ONE_MINUTE));
+    const s = Math.floor(normalizeAge((ms % ONE_MINUTE) / ONE_SECOND));
+
+    return `+ ${d > 0 ? d + "d " : ""}${h > 0 ? h + "h " : ""}${ m > 0 ? m + "m " : ""}${s > 0 ? s + "s " : ""}`;
```
Can you create another utility function `addTimeSuffix(value, suffix) {...}` where you pass in the number and the suffix, it'll check if it's greater than zero, then return the value plus suffix if so or just the value if not. So for days it would be: `const d = addTimeSuffix(Math.floor(normalizeAge(ms / ONE_DAY)), 'd ');` then you don't have to user the tertiary operators in the returned string.

jvigliotta reviewed (changes requested):
This looks (and works) great! I do have a couple requests related to making things a little more readable and deliberate. I've included those inline.

jvigliotta commented on src/utils/duration.js:
```diff
@@ -39,6 +40,22 @@ function toDoubleDigits(num) {
+        addTimeSuffix(Math.floor(normalizeAge((ms % ONE_DAY) / ONE_HOUR)), 'h'),
+        addTimeSuffix(Math.floor(normalizeAge((ms % ONE_HOUR) / ONE_MINUTE)), 'm'),
+        addTimeSuffix(Math.floor(normalizeAge((ms % ONE_MINUTE) / ONE_SECOND)), 's')
+    ].filter(Boolean).join(' ');
```
Can we remove this filter? I don't think it's necessary. Otherwise, looks great!

mariuszr replied:
Added missing an extra "l" in the method name - should be 'millisecondsToDHMS' instead of 'milisecondsToDHMS'.

I've added the missed extra 'l'. As to the array.filter(Boolean), the filter is the same as array.filter(i => Boolean(i)) which in this case, filters some of DHMS elements that are empty string and generate extra space during joining.

jvigliotta replied:
Ah... gotcha! Nice! I'm going to get one more set of eyes on this as the formatting for UTC is widely used. Otherwise looks great!

jvigliotta reviewed (changes requested):
This is looking really good, I have one code related change inline. Also, can you change `milisecondsToDHMS` to `millisecondsToDHMS` wherever it's used (just missing an extra "l").

mariuszr commented:
Hi @davetsay, 
1-2. Not a problem, I have changed it however, please note that the UTCTimeFormat.js is located in the part of the app (platform) that follows the ES5 (most likely the code is before converting to ES6), and converting partial to ES6 may confused developer. We can also see, that [EsLint](https://github.com/nasa/openmct/blob/master/.eslintrc.js#L265-L266) does not consider these parts of codes by overriding rules for `no-var` and `one-var`.
3. Seems like the `getDuration` method is not used in other places so I have removed it

davetsay reviewed (approved):
LGTM.

There were some lint errors but they were not yours. The lint errors have been resolved in another PR so this looks good to go.

## Reviewer Checklist
| | |
| --- | :---: |
| Changes appear to address issue? | Y |
| Appropriate unit tests included? | N/A |
| Code style and in-line documentation are appropriate? | Y |
| Commit messages meet standards? | Y |
| Has associated issue been labelled `unverified`? | Y |
